File: core-java/src/test/java/com/baeldung/datetime/UseZonedDateTimeUnitTest.java
package com.baeldung.datetime;

import java.time.LocalDateTime;
import java.time.ZoneId;
import java.time.ZonedDateTime;

import org.junit.Assert;
import org.junit.Test;

public class UseZonedDateTimeUnitTest {

    UseZonedDateTime zonedDateTime = new UseZonedDateTime();

    @Test
    public void givenZoneId_thenZonedDateTime() {
        ZoneId zoneId = ZoneId.of("Europe/Paris");
        ZonedDateTime zonedDatetime = zonedDateTime.getZonedDateTime(LocalDateTime.parse("2016-05-20T06:30"), zoneId);
        Assert.assertEquals(zoneId, ZoneId.from(zonedDatetime));
    }
}

File: core-kotlin-modules/core-kotlin-2/src/test/kotlin/com/baeldung/dates/datetime/UseZonedDateTimeUnitTest.kt
package com.baeldung.kotlin.datetime

import com.baeldung.dates.datetime.UseZonedDateTime
import org.junit.Assert
import org.junit.Test
import java.time.LocalDateTime
import java.time.ZoneId

class UseZonedDateTimeUnitTest {

    internal var zonedDateTime = UseZonedDateTime()

    @Test
    fun givenZoneId_thenZonedDateTime() {
        val zoneId = ZoneId.of("Europe/Paris")
        val zonedDatetime = zonedDateTime.getZonedDateTime(LocalDateTime.parse("2016-05-20T06:30"), zoneId)
        Assert.assertEquals(zoneId, ZoneId.from(zonedDatetime))
    }
}